akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileUploadDirectives.scala

```diff
@@ -85,6 +117,35 @@ trait FileUploadDirectives {
       case Some(tuple) ⇒ provide(tuple)
       case None        ⇒ reject(MissingFormFieldRejection(fieldName))
     }
+
+  /**
+   * Collects each body part that is a multipart file as a tuple containing metadata and a `Source`
+   * for streaming the file contents somewhere. If there is no such field the request will be rejected.
+   * Files are buffered into temporary files on disk so in-memory buffers don't overflow.
+   *
+   * @group fileupload
+   */
+  def fileMultiUpload(fieldName: String): Directive1[immutable.Seq[(FileInfo, Source[ByteString, Any])]] =
+    extractRequestContext.flatMap { ctx ⇒
+      implicit val mat = ctx.materializer
+      implicit val ec = ctx.executionContext
+
+      uploadedFiles(fieldName).flatMap { files ⇒
+        val partsSource: Source[(FileInfo, Source[ByteString, Any]), Any] = Source(files).map {
```

Comment:
This can be even simpler: no need for `Source(files)` when source and target datastructures are a `Seq`. You can just map the `files` datastructure.